src/test/scala/org/ergoplatform/nodeView/state/ErgoStateContextSpec.scala

```diff
@@ -73,13 +87,23 @@ class ErgoStateContextSpec extends HistoryTestHelpers {
     ).fields
     sc.appendFullBlock(fbWithFields(invalidInterlinks ++ oldFields)) shouldBe 'failure
 
+    // https://github.com/ergoplatform/ergo/issues/2114
+    // if validMKV._1.head is 1, appendFullBlock within "valid application of correct extension" will fail
+    // As a workaround, just generate new values until (validMKV._1.head != 1)
+    // TODO: investigate and provide a full fix (followup issue)
```

Comment:
this fix is okay here, just add a note that with key.head == 1 improperly packed interlink would be generated

Reply:
added